obsplus/utils/bank.py

```diff
@@ -261,15 +228,28 @@ def clear_cache(self):
 
 @contextlib.contextmanager
 def sql_connection(path, **kwargs):
-    con = sqlite3.connect(path, **kwargs)
+    """
+    A simple context manager for a sqlite connection.
+
+    Parameters
+    ----------
+    path
+        A path to the sqlite database.
+    """
+    con = sqlite3.connect(str(path), **kwargs)
     with con:
         yield con
+    con.close()  # this is needed on windows but not linux, weird...
 
 
 def get_kernel_query(starttime: int, endtime: int, buffer: int):
-    """" get the HDF5 kernel query parameters (this is necessary because
-    hdf5 doesnt accept invert conditions for some reason. A slight buffer
-    is applied to the ranges to make sure no edge files are left out"""
+    """
+    Create a HDF5 kernel query based on start and end times.
+
+    This is necessary because hdf5 doesnt accept inverted conditions.
+    A slight buffer is applied to the ranges to make sure no edge files
+    are excluded.
+    """
```

Comment:
This probably won't be used by many end-users, but because it's not private, is it worth adding a quick params section to the doc-string?

Reply:
Good point! Looking at it, there is no reason for this function to be public, so I will just make it private.